File: icl_backend/src/main/java/es/uoc/icl/domain/Negocio.java
package es.uoc.icl.domain;

import java.math.BigDecimal;
import java.util.Collection;
import java.util.List;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import lombok.Data;

@Data
public class Negocio implements UserDetails{

	private static final long serialVersionUID = 1L;
	private Integer id;
    private String cif;
    private String email;
    private String nombre;
    private TipoNegocio tipo;
    private String direccion;
    private BigDecimal valorTotal;
    private String contrasena;
    private Integer estado;
    private BigDecimal totalCanjeado;
    
	@Override
	public Collection<? extends GrantedAuthority> getAuthorities() {
		return List.of(new SimpleGrantedAuthority(Rol.NEGOCIO.name()));
	}
	@Override
	public String getPassword() {
		return contrasena;
	}
	@Override
	public String getUsername() {
		return email;
	}
	@Override
	public boolean isAccountNonExpired() {
		return true;
	}
	@Override
	public boolean isAccountNonLocked() {
		return true;
	}
	@Override
	public boolean isCredentialsNonExpired() {
		return true;
	}
	@Override
	public boolean isEnabled() {
		return true;
	}
}

File: icl_frontend/src/app/models/negocio.ts

import { TipoNegocio } from "./tipo-negocio";

export class Negocio {
  id: string;
  nombre: string;
  direccion: string;
  cif: string;
  email: string;
  contrasena: string;
  valorTotal: number;
  estado: number;
  tipo: TipoNegocio;
  totalCanjeado: number;

  constructor(
    id: string = '',
    nombre: string = '',
    direccion: string = '',
    cif: string = '',
    email: string = '',
    contrasena: string = '',
    valorTotal: number = 0,
    estado: number = 1,
    tipo: TipoNegocio = new TipoNegocio,
    totalCanjeado: number = 0
  ) {
    this.id = id;
    this.nombre = nombre;
    this.direccion = direccion;
    this.cif = cif;
    this.email = email;
    this.contrasena = contrasena;
    this.valorTotal = valorTotal;
    this.estado = estado;
    this.tipo = tipo;
    this.totalCanjeado = totalCanjeado;
  }
}